components/brave_ads/browser/ads_service.h

```diff
@@ -19,14 +20,44 @@
 #include "brave/components/services/bat_ads/public/interfaces/bat_ads.mojom.h"
 #include "components/keyed_service/core/keyed_service.h"
 #include "mojo/public/cpp/bindings/pending_remote.h"
+#include "ui/message_center/public/cpp/notification.h"
 
 class GURL;
 
 namespace brave_ads {
 
 class AdsService : public KeyedService {
  public:
-  AdsService();
+  class Delegate {
+   public:
+    virtual void InitNotificationHelper() = 0;
+    virtual bool CanShowSystemNotificationsWhileBrowserIsBackgrounded() = 0;
+    virtual bool DoesSupportSystemNotifications() = 0;
+    virtual bool CanShowNotifications() = 0;
+    virtual bool ShowOnboardingNotification() = 0;
+    virtual void ShowScheduledCaptcha(const std::string& payment_id,
+                                      const std::string& captcha_id) = 0;
+    virtual void ClearScheduledCaptcha() = 0;
+    virtual void SnoozeScheduledCaptcha() = 0;
+    virtual void Display(const message_center::Notification& notification) = 0;
+    virtual void Close(const std::string& notification_id) = 0;
+    virtual void ShowNotificationAd(const std::string& id,
+                                    const std::u16string& title,
+                                    const std::u16string& body);
```

Comment:
Missed = 0? I've got 'undefined symbol' referenced by 'const brave_ads::AdsService::Delegate::`vftable'

Reply:
@cdesouza-chromium says he is creating a PR for it

Reply:
@boocmp what build were you running when you saw that?

Reply:
I've already merged a fix. I should have linked to it https://github.com/brave/brave-core/pull/25847

Reply:
I believe it may have been debug builds only. At least, that's what it seemed like when I encountered it.

Reply:
Yes, it only fails when optimization is turned off